### DISCUSSION TITLE
What are the models/tools we want to wrap?

### DISCUSSION THREAD

**Discussion by bethac07** [bilayer-containers/bilayers#3] [Running lists of things]
Running unordered list, to be turned into issues and sorted in projects and milestone

**Comment by bethac07** [comment]
Cellpose

**Reply by bethac07** [reply]
@rajavishah @gnodar01 I updated the couple of containers we already know we want [here](https://docs.google.com/spreadsheets/d/10tnusZYhi0wXXroPUgdGUgxf6xsDaBoPpFjC7eObTlo/edit?pli=1&gid=0#gid=0) - I know we wanted to start with Cellpose, so putting it in the comments here. I strongly suggest, though, that after you do this second algorithm, that then the next thing to do is to write up instructions on adding algorithms, so that we can ask the image analysts to add 3rd, 4th, 5th, etc.

**Reply by bethac07** [reply]
Done now for 2 (3 and 4 just waiting for algorithm versioning, though how we want to handle multi-major-version support is still a thing I guess we'll need to have a plan for!)

**Comment by bethac07** [comment]
StarDist

**Reply by bethac07** [reply]
Done!

**Comment by bethac07** [comment]
DeepProfiler

**Comment by bethac07** [comment]
ilastik

**Comment by bethac07** [comment]
Florian recommends we try to do [EmbedSeg](https://juglab.github.io/EmbedSeg/)

**Reply by helia77** [reply]
The pre-trained models on the webpage are no longer available. We also couldn't find them on the GitHub repo. Worth contacting them directly.

**Reply by bethac07** [reply]
https://github.com/juglab/EmbedSeg/issues/38#issuecomment-4912614709

**Comment by bethac07** [comment]
[Ultrack](https://www.biorxiv.org/content/10.1101/2024.09.02.610652v1.full.pdf+html)

**Comment by bethac07** [comment]
Instanseg https://x.com/petebankhead/status/1833945619481985405?t=5a57u8pFuedCxsL4iY3HlQ&s=19

**Reply by bethac07** [reply]
https://doi.org/10.1101/2024.09.04.611150

**Reply by gnodar01** [reply]
> It's much faster than other cell segmentation methods

😲

>  irrespective of the number or ordering of imaged biomarkers.

😲😲

> InstanSeg can be entirely (including prostprocessing!) compiled in TorchScript

😲😲😲

**Reply by bethac07** [reply]
Ikr?  And Pete isn't the sort of guy to make wild claims he can't back up.

Let's just... do it, man. It's already got a well documented CLI, seemingly, too.

**Reply by bethac07** [reply]
Done, though we might want to bump versions

**Comment by bethac07** [comment]
@jccaicedo has a micronucelus detector he thinks may suit

**Reply by bethac07** [reply]
https://huggingface.co/CaicedoLab/mnDINO

**Comment by bethac07** [comment]
MicroSAMs auto-segmenter!

**Reply by bethac07** [reply]
I tried this today, initial attempts (<1hr live demo) failed because it wants a specific file to be pointed to, not a folder with files in it. I'm not sure how we deal with this; it might be that I just need to read the spec more closely, OR we adapt their CLI, OR we say "not compatible".

Something someone suggested, and I'm open to _theoretically but initially resistantly_ is the idea that if we pass in an input folder, there's some flag that says "iterate on all of these files, rather than pass in the folder" - I think it would depend on how often we see that behavior pattern.

**Reply by rajavishah** [reply]
Here #175

**Reply by bethac07** [reply]
> it might be that I just need to read the spec more closely

Wins every time! Thanks for the nice find :)

**Reply by rajavishah** [reply]
> it might be that I just need to read the spec more closely

I'm glad then – I was a bit disheartened that live demo didn’t go as planned :(

**Reply by bethac07** [reply]
I mean, to be clear, it was a live demo to a room full of other people who develop software, who I was very emphatic with that I had _in no way prepped to do this,_ who were all super impressed that all the interface generation worked out of the box! It just didn't actually run the MicroSAM algorithm all the way through. But I'm making sure to relay right now in the hackathon discussion channel that you solved it already because you're a freaking rockstar!

**Reply by rajavishah** [reply]
Haha, thanks! That's a bit relief 😄!

**Comment by bethac07** [comment]
[MegaSeg](https://github.com/AllenCell/allencell-segmenter-ml?tab=readme-ov-file#models)

**Comment by bethac07** [comment]
[CellSAM](https://github.com/vanvalenlab/cellSAM)

**Reply by helia77** [reply]
The model weights are gated through DeepCell. It requires to create an account (at https://users.deepcell.org) and generate a token. In progress!

**Reply by helia77** [reply]
Done.
The users have to provide their own access token and pass through the CLI.

**Comment by bethac07** [comment]
Uri Manor has some virtual staining models he'd be excited to include

**Comment by saramcardle** [comment]
MCMICRO. Both the install and file setup scare me. I'm personally most interested in the Ashlar stitching portion, but I think others might care more about the segmentation.  
https://mcmicro.org/

**Comment by bethac07** [comment]
@jccaicedo 's new CHAMMI-75 models

**Reply by bethac07** [reply]
https://huggingface.co/CaicedoLab/MorphEm

**Reply by bethac07** [reply]
These are the feature ones!

**Reply by helia77** [reply]
Done!

**Comment by bethac07** [comment]
BIMZ ModelZoo Models (discussions going back to original I2K, just forgot to put here apparently!)
